crypto/ec/ecp_nistz256.c

```diff
@@ -1357,9 +1402,109 @@ __owur static int ecp_nistz256_points_mul(const EC_GROUP *group,
+                                            0xf,0xc,0x6,0x3,
+                                            0x2,0x5,0x4,0xf};
+
+    BN_ULONG table[P256_LIMBS*15];
+    BN_ULONG out[P256_LIMBS], t[P256_LIMBS];
+    int i, ret = 0;
+    BIGNUM *tmp;
+
+    if ((BN_num_bits(x) > 256)
+        || BN_is_negative(x)) {
+        if ((tmp = BN_CTX_get(ctx)) == NULL) {
+            ECerr(EC_F_ECP_NISTZ256_INV_ORD, ERR_R_BN_LIB);
+            goto err;
+        }
+        if (!BN_nnmod(tmp, x, group->order, ctx)) {
+            ECerr(EC_F_ECP_NISTZ256_INV_ORD, ERR_R_BN_LIB);
+            goto err;
+        }
+        x = tmp;
+    }
+    /* We don't use entry 0 in the table, so we address with -1 offset */
+    ecp_nistz256_bignum_to_field_elem(out, x);
+    ecp_nistz256_ord_mul_mont(&table[0*P256_LIMBS], out, RR);
+    for ( i = 2; i < 16; i+=2 ) {
+        ecp_nistz256_ord_sqr_mont(&table[(i-1)*P256_LIMBS],
+                                  &table[(i/2-1)*P256_LIMBS], 1);
+        ecp_nistz256_ord_mul_mont(&table[i*P256_LIMBS],
+                                  &table[(i-1)*P256_LIMBS],
+                                  &table[0*P256_LIMBS]);
+    }
+    /* The top 128bit of the exponent are highly redundant,
+       so we perform an optimized flow */
+    /* f */
+    memcpy(out, &table[(15-1)*P256_LIMBS], sizeof(out));
+    /* f0 */
+    ecp_nistz256_ord_sqr_mont(out, out, 4);
+    /* ff */
+    ecp_nistz256_ord_mul_mont(out, out, &table[(15-1)*P256_LIMBS]);
+    memcpy(t, out, sizeof(t));
+    /* ff00 */
+    ecp_nistz256_ord_sqr_mont(out, out, 8);
+    /* ffff */
+    ecp_nistz256_ord_mul_mont(out, out, t);
+    memcpy(t, out, sizeof(t));
+    /* ffff0000 */
+    ecp_nistz256_ord_sqr_mont(out, out, 16);
+    /* ffffffff */
+    ecp_nistz256_ord_mul_mont(out, out, t);
+    memcpy(t, out, sizeof(t));
+    /* ffffffff0000000000000000 */
+    ecp_nistz256_ord_sqr_mont(out, out, 64);
+    /* ffffffff00000000ffffffff */
+    ecp_nistz256_ord_mul_mont(out, out, t);
+    /* ffffffff00000000ffffffff00000000 */
+    ecp_nistz256_ord_sqr_mont(out, out, 32);
+    /* ffffffff00000000ffffffffffffffff */
+    ecp_nistz256_ord_mul_mont(out, out, t);
+
+    /* The bottom 128 bit of the exponent are easier done with a table */
+    for( i = 0; i < 32; i++ ) {
+        ecp_nistz256_ord_sqr_mont(out, out, 4);
+        /* The exponent is public, no need in constant time access */
+        ecp_nistz256_ord_mul_mont(out, out, &table[(expLo[i]-1)*P256_LIMBS]);
+    }
+    ecp_nistz256_ord_mul_mont(out, out, CONST_ONE);
+
+    if (!bn_set_words(r, out, P256_LIMBS)) {
+        ECerr(EC_F_ECP_NISTZ256_INV_ORD, ERR_R_BN_LIB);
+        goto err;
+    }
+    ret = 1;
+err:
+    return ret;
+}
+#endif
+
 __owur static int ecp_nistz256_get_affine(const EC_GROUP *group,
-                                          const EC_POINT *point,
-                                          BIGNUM *x, BIGNUM *y, BN_CTX *ctx)
+                                   const EC_POINT *point,
```

Comment:
Nit: the C has wrong indentation/spacing/style in places - if you could do a quick pass while rebasing, that'd be great. E.g. `lowercase_with_underscore` naming, spaces around assignment and comparison `a *= 2`, no spaces inside parens `(i = 0; i < 32; i++)` etc. (coding style: https://www.openssl.org/policies/codingstyle.html).